app/controllers/VaultController.scala

```diff
@@ -227,4 +227,85 @@ class VaultController @Inject() (cc:ControllerComponents,
       }
     }
   }
-}
+
+  def loadExistingContent(vaultId: String) = {
+    val interestingFields = Array(
+      "MXFS_PATH",
+      "MXFS_FILENAME",
+      "GNM_ASSET_FOLDER",
+      "GNM_TYPE",
+      "GNM_PROJECT_ID"
+    )
+
+    val catchAllSearchTerm = SearchTerm.createNOTTerm(SearchTerm.createSimpleTerm("oid", ""))
+    val finalSink = Sink.seq[CachedEntry]
+    val content = userInfoCache.infoForVaultId(vaultId)
+    //implicit val vault:Vault = MatrixStore.openVault(content.head)
+    val graph = GraphDSL.create(finalSink) { implicit builder =>
+      sink =>
+        import akka.stream.scaladsl.GraphDSL.Implicits._
+
+        val src = builder.add(new OMFastSearchSource(content.head, Array(catchAllSearchTerm), interestingFields, atOnce = 100))
+        src.out.map(elem => {
+          //val mxsEntry = vault.getObject(elem.oid)
+          //val checkSum = MetadataHelper.getOMFileMd5(mxsEntry) match {
+          //  case Failure(err)=>
+          //    logger.warn(s"Could not get appliance MD5: ", err)
+          //    "None"
+          //  case Success(checksum)=>checksum
+          //}
+          val ent = CachedEntry(
+            elem.oid,
+            elem.attributes.flatMap(_.stringValues.get("MXFS_PATH")).getOrElse("(no path)"),
+            elem.attributes.flatMap(_.stringValues.get("MXFS_FILENAME")).getOrElse("(no filename)"),
+            elem.attributes.flatMap(_.stringValues.get("GNM_ASSET_FOLDER")),
+            elem.attributes.flatMap(_.stringValues.get("GNM_TYPE")),
+            elem.attributes.flatMap(_.stringValues.get("GNM_PROJECT_ID")),
+            ""
+          )
+
+          logger.debug(s"Got entry $ent")
+          ent
+        }) ~> sink
+        ClosedShape
+    }
+
+    RunnableGraph.fromGraph(graph).run()
+  }
+
+  case class FullDuplicateData(mxfsPath:String, duplicateNumber:Int, duplicatesData:Seq[CachedEntry])
+  case class AllDuplicateData(dupes_count:Int, item_count:Int, duplicates:Array[FullDuplicateData])
+
+  def getDuplicateData(vaultId: String)  = {
+    loadExistingContent(vaultId).map(results=>{
+      val contentCache = new ExistingArchiveContentCache(results)
+      var duplicatesArray: Array[FullDuplicateData] = Array()
+      val dupeCount = contentCache.dupesCount
+      if (dupeCount > 0) {
+        logger.warn(s"There are $dupeCount duplicated files in the archive")
+        contentCache.dupedPaths.foreach(dupe => {
+          logger.debug(s"${dupe._1}: ${dupe._2} copies")
+          val duplicatedItemData = contentCache.getAllForPath(dupe._1)
+          val duplicateDataThree = FullDuplicateData(dupe._1,dupe._2,duplicatedItemData)
+          duplicatesArray +:= duplicateDataThree
+        })
+      } else {
+        logger.info(s"No duplicates found.")
+      }
```

Comment:
(and line 282)

You should not be using a mutable array here (in Scala that is).
Instead you should write:

```scala
val duplicatesArray = contentCache.dupedPaths.map(dupe=>{
  logger.debug(s"${dupe._1}: ${dupe._2} copies")
  val duplicatedItemData = contentCache.getAllForPath(dupe._1)
  FullDuplicateData(dupe._1, dupe._2, duplicatedItemData)
})
```

Which achieves the same result (empty sequence if contentCache.dupesCount is 0, otherwise a sequence of `FullDuplicateData` corresponding to the items in `contentCache.dupedPaths`) and avoids the discouraged use of mutable sequences.

Reply:
Your suggested code does not work so I am going to leave the existing code.

Reply:
I have changed the code based on your futurer suggestions and successfully tested it.